src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java

```diff
@@ -101,7 +106,19 @@ public String getPartitioner() {
 
   @Override
   public String getClusterName() {
-    return null; // TODO: implement me.
+    // Cluster name is part of Endpoint States.
+    try {
+      EndpointStates endpoints = apiClient.getEndpointStates();
+      assert endpoints != null;
+      List<Map<String, String>> states = endpoints.getEntity();
+      assert states != null;
+      assert !states.isEmpty();
+      // cluster name should be the same for all states
+      return states.get(0).get("CLUSTER_NAME");
+    } catch (ApiException ae) {
+      LOG.error("Failed to retrieve Cluster Name from Endpoint states data", ae);
+    }
+    return null;
```

Comment:
thought: I wonder if we shouldn't throw an exception here (it would have to be an unchecked one since the method doesn't declare `ReaperException`).
Looking at the method usages, there is at least one case where null is not supported, in `Cluster.Builder.build()`.

Reply:
Good point. When I first coded this, I wasn't sure what we could/should do if we can't get the cluster name. Stuff will break for sure.

Reply:
Question: @Miles-Garnsey do you think we could/should define the parent interface such that it declares a throw of  ReaperException here?

Reply:
I don't think so, logging is probably sufficient. If we throw an exception all we'd be doing is logging it and retrying I think, since there's always a chance that the next response will be right. 

However, you may want to include the endpoint states data that was bad in the log.

Reply:
Yeah, I think it's just going to be an NPE down the path. I'll add what I can to the exception log.

Reply:
I don't know what else we can do to be honest. I always think there's not much point throwing an exception if the caller isn't able to handle it anyway, and I'm not sure (beyond a retry) what the caller can do in this case.

Reply:
So I can't really add much to the catch of the `ApiException`. It will only be thrown if the call to getEndpoints fails. I can log things along the way if that call succeeds, but we still are missing things we expect in the response.

Reply:
@Miles-Garnsey I think I have covered the cases here.....

Reply:
I think that's fine! A bit tough to review this one as I think it combines a number of commits from other PRs. Can we get those merged, rebase and then do a final review?